xds/internal/client/xds.go

```diff
@@ -571,62 +571,54 @@ func unmarshalClusterResource(r *anypb.Any, logger *grpclog.PrefixLogger) (strin
 	return cluster.GetName(), cu, nil
 }
 
-func clusterTypeFromCluster(cluster *v3clusterpb.Cluster) (ClusterType, string, []string, error) {
+func validateClusterAndConstructClusterUpdate(cluster *v3clusterpb.Cluster) (ClusterUpdate, error) {
+	if cluster.GetLbPolicy() != v3clusterpb.Cluster_ROUND_ROBIN {
+		return ClusterUpdate{}, fmt.Errorf("unexpected lbPolicy %v in response: %+v", cluster.GetLbPolicy(), cluster)
+	}
+
+	// Process security configuration received from the control plane iff the
+	// corresponding environment variable is set.
+	var sc *SecurityConfig
+	if env.ClientSideSecuritySupport {
+		var err error
+		if sc, err = securityConfigFromCluster(cluster); err != nil {
+			return ClusterUpdate{}, err
+		}
+	}
+
+	ret := ClusterUpdate{
+		ClusterName: cluster.GetName(),
+		EnableLRS:   cluster.GetLrsServer().GetSelf() != nil,
+		SecurityCfg: sc,
+		MaxRequests: circuitBreakersFromCluster(cluster),
+	}
+
+	// Validate and set cluster type from the response.
```

Comment:
Does it make more sense to use a `switch` instead? I believe all these `if`s are mutually exclusive, right?

Reply:
Done.